org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLCompletionSnippetsTest.java

```diff
@@ -92,12 +108,19 @@ public void emptyXMLContent() throws BadLocationException {
 				c("Insert XML Processing Instruction", //
 						"<?xml version=\"1.0\" encoding=\"UTF-8\"?>", //
 						r(0, 0, 0, 2), "<?xml"),
+				c("Insert XML Model Processing Instruction to Associate XSD Schema", //
+						"<?xml-model href=\"file.xsd\" type=\"application/xml\" schematypens=\"http://www.w3.org/2001/XMLSchema\"?>", //
+						r(0, 0, 0, 2), "<?xml-model"),
+				c("Insert XML Model Processing Instruction to Associate Document Type Description (DTD)", //
+						"<?xml-model href=\"file.dtd\" type=\"application/xml-dtd\"?>", //
+						r(0, 0, 0, 2), "<?xml-model"),
 				c("<!--", //
 						"<!-- -->", //
 						r(0, 0, 0, 2), "<!--"));
 
-		testCompletionFor("<!|", NEW_XML_SNIPPETS /* DOCTYPE snippets */ + //
+		testCompletionFor("<!|", NEW_XML_SNIPPETS /* DOCTYPE snippets */ + // TODO: why does that trigger these completions? is that intended?

```

Comment:
Yes we return all snippets and filter is done by client. It gives the capability to benefit with fuzzy completion if client (like vscode, Eclipse IDE) supports it.

Reply:
ok :)